### ISSUE TITLE
Multiple operations failing with: Post "https://sts.amazonaws.com/": net/http: TLS handshake timeout

### ISSUE THREAD

**Issue by soulshake** [hashicorp/terraform-provider-aws#19264]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
$ terraform version
Terraform v0.15.2
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v3.38.0
# ...
```

### Affected Resource(s)

n/a, occurs even on `terraform init`

### Terraform Configuration Files

```hcl
provider "aws" { 
  allowed_account_ids = [redacted]
  region              = "us-east-2"
}

terraform {
  backend "s3" {
    bucket               = "terraform-state-redacted"
    dynamodb_table       = "client-redacted-infra-terraform-state-locks"
    encrypt              = true
    key                  = "tfstate"
    region               = "us-east-2"
  }
}

terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 3.37.0"
    }
  # ...
}
```

### Debug Output

See [gist](https://gist.github.com/soulshake/62ecbf057a023234b33f31cfdb1d2394)

### Expected Behavior

Error should not occur

### Actual Behavior

Various Terraform operations (`init`, `plan`, even `show` and `workspace select`) fail sporadically but frequently in CI with: 

```
  2021-05-06T09:59:01.845Z [TRACE] Meta.Backend: using already-initialized, unchanged "s3" backend configuration
  2021-05-06T09:59:01.846Z [INFO]  Attempting to use session-derived credentials
  ╷
  │ Error: error configuring S3 Backend: no valid credential sources for S3 Backend found.
  │ 
  │ Please see https://www.terraform.io/docs/language/settings/backends/s3.html
  │ for more information about providing credentials.
  │ 
  │ Error: RequestError: send request failed
  │ caused by: Post "https://sts.amazonaws.com/": net/http: TLS handshake timeout
```

### Steps to Reproduce

I haven't found a way to reproduce reliably, but for us:

1. Run `terraform init` in parallel on about 17 subdirectories

### Important Factoids

- We're running about 17 parallel `init`/`workspace select`/`plan`/`show` operations on different Terraform configurations in CI (Github actions).
- Restarting the workflow sometimes allows the failed job to succeed, but other jobs may fail with the same error.
- We happen to be in the middle of a migration to self-hosted Github actions runners, and have also observed this behavior on self-hosted runners.

I have tried various combinations of:

##### in env

-  `AWS_MAX_ATTEMPTS: 5`
-  `AWS_RETRY_MODE: adaptive`
-  `AWS_STS_REGIONAL_ENDPOINTS: regional`
- `TF_CLI_ARGS_apply: -lock-timeout=420s -parallelism=1`
- `TF_CLI_ARGS_init: -parallelism=1`
- `TF_CLI_ARGS_plan: -lock-timeout=420s -parallelism=1`
- `TF_CLI_ARGS_show: -parallelism=1`

##### in `aws` provider block
- `  max_retries         = 50` 

##### in `backend "s3"` block
- `max_retries                 = 10` 
- `skip_credentials_validation = true`

... all with no apparent effect.

I've also tried adding a random sleep up to 30 seconds before the first Terraform operation in a given CI job, without success.

The only thing that seems to make the error less likely is if fewer jobs are run in parallel, but I'm not 100% sure that truly makes a difference.

### References

- https://github.com/hashicorp/terraform-provider-aws/issues/14163 (the closest thing I've found, but seems unlikely to be related)

**Comment by ewbankkit** [comment]
@soulshake Thanks for raising this issue.
The Terraform S3 Backend is part of the Terraform CLI, not the AWS Provider.
Could you please raise an issues in their repository: https://github.com/hashicorp/terraform?

**Comment by soulshake** [comment]
Good point! My mistake, thanks <3

**Comment by Cajga** [comment]
@soulshake, we got the same issue, have you opened the case in the terraform repo?

**Comment by Cajga** [comment]
I opened a new issue there: https://github.com/hashicorp/terraform/issues/28714
